compiler/src/vm/heap/mod.rs

```diff
@@ -125,50 +128,54 @@ impl Heap {
 
     /// Clones all objects at the `root_addresses` into the `other` heap and
     /// returns a list of their addresses in the other heap.
-    pub fn clone_multiple_to_other_heap(
+    pub fn clone_multiple_to_other_heap_with_existing_mapping(
         &self,
         other: &mut Heap,
         addresses: &[Pointer],
+        address_map: &mut HashMap<Pointer, Pointer>,
```

Comment:
Oh, right, I didn't think of that.

We'll definitely need to look into that then. We could of course look into using a deduplicating heap for the tracing so that even though the objects are fully traversed every time, only one instance is stored by the tracer. But I think it's fine that we do that later when we introduce a better heap representation than a `HashMap`.

Do you agree or do you think I should address this now?

Reply:
No, we can address this later when it becomes relevant. Deduplication in general sounds good since it could also improve fiber communication, but, in this case, we could also check whether the value is the same as we previously copied from that address or we could add a generation counter for the values behind an address. But we can decide on that later on